### ISSUE TITLE
1.8 update checklist

### ISSUE THREAD

**Issue by luacs1998** [ForgeEssentials/ForgeEssentials#1141]
Do we want to do this now? I'd prefer having a stable 1.7.10 release first though, but then again I like to be on the cutting edge...

Will just leave this checklist here first so we know what to do. Will be updated when more info on 1.8 comes out.
## 1.7 -> 1.8

- [x] All FML imports moved to net.minecraftforge
- [x] Typical MC architecture changes (only those which affect us has been posted):
 - Almost all int x, int y, int z parameters have been changed to BlockPos class
 - ForgeDirection has been removed, replaced by net.minecraft.util.EnumFacing.
 - Metadata has been extracted out in Minecraft to IBlockState, which holds a list of properties instead of magic number metadata. DO NOT listen to the fearmongering, you can do EVERYTHING with block states you could previously with metadata.
- [ ] Update mixins (#1619)
 - [x] EntityPlayer_01
 - [ ] BlockPortal_01
 - [x] NetHandlerPlayServer_01 (needs updating Forge extensions)
 - [ ] NetHandlerPlayServer_02
 - [x] CommandHandler_01
 - [x] EntityTracker_01
 - [x] Entity_01 
 - [x] ItemInWorldManager_01
- [ ] Modify our worldborder to be compatible with MC's
- [ ] Settle sponge things #1031 
- [ ] Remove our commands that vanilla adds (doas)

## 1.8 -> 1.8.9

- [ ] Port to 1.8.9
- [ ] Update to FG2
- [ ] Java 8 compatibility

**Comment by TheBeege** [comment]
@luacs1998 Are folks interested in resuming this? The server that I work with needs to upgrade to 1.8 to support the primary mod we're using, but it also relies on ForgeEssentials. I'm willing to contribute, but I'm curious as to the larger community's stance on this. I'll probably end up bugging you guys in IRC.

**Comment by olee** [comment]
1.8 update is planned for April if everything goes fine.
Currently I'm very busy, but in a week I will have time again and will try to finish all that's left for 1.7.10 so we can move on to 1.8.

**Comment by longlostbro** [comment]
Is there an update on this estimated date?

**Comment by olee** [comment]
We still need a bit more time to get out of beta for 1.7.10.
If we update too early, we will not be able to support both 1.7.10 and 1.8 at the same time.
Currently I am rewriting a lot of old code to get FE ready for an easy update process and then be able to maintain both 1.7.10 and 1.8 at the same.

TL;DR: Wait one or two weeks and see if we can get a non-beta release out. After that we will (probably) start updating.

**Comment by luacs1998** [comment]
Very likely we may be doing parallel releases for 1.7.10 and 1.8.

I'll put olee in charge of the 1.7.10 releases, and do the 1.8 update myself.

We'll need to configure jenkins to build multiple branches though.

**Comment by liach** [comment]
Nice!!!!!!

**Comment by olee** [comment]
TL;DR: I code and @luacs1998 merges changes to 1.8? :smile:

**Comment by luacs1998** [comment]
Yes, but I would still prefer it if you discussed major and not-so-major stuff with me first.

**Comment by liach** [comment]
So will there be a branch for 1.8 now, as 1.4.0 is released?

**Comment by olee** [comment]
Yeah probably - though I don't have much time right now, I will help out with as well whenever I have time.

**Comment by olee** [comment]
Got working 1.8 branch now - closing.

**Comment by luacs1998** [comment]
Working doesn't mean checklist is complete.

**Comment by luacs1998** [comment]
Also, the separate job for 1.8 builds should be rolled into the main job when we decide to shift all development efforts to 1.8.

**Comment by olee** [comment]
Well 1.7.10 is kinda a LTS version so this will probably take at least half a year before we can consider removing that job.

**Comment by luacs1998** [comment]
1.6.4 was also LTS too until recently and Jenkins only built 1.7 jobs.

if there's a need to make emergency bugfix releases we can always build manually.

**Comment by olee** [comment]
Though keeping the job active makes these things easier.

**Comment by luacs1998** [comment]
Then it gets hived off.

**Comment by luacs1998** [comment]
I'll probably work on the 1.8.9 port one of these days.

What this means is that the 1.8.0 versions will no longer be supported, and our supported 1.8.x version will go up to 1.8.9.

Expect to see a final 1.8.0 build on CurseForge soon.

@olee - this will also be a test of how version-proof your patcher is :D

**Comment by liach** [comment]
When will Fe support java8 only?

**Comment by luacs1998** [comment]
My plan is 1.8.9 update.

But plans can and do change.

**Comment by DPxKnightxZero** [comment]
1.8.9 has a massive performance increase! What used to take minutes to load a 32 size of chunks now takes about 10 secs. from 1.7.10 that is a 10000% increase in performance. Even loading new chunks does not drag my fps down I still remain at 57 - 62

**Comment by olee** [comment]
1.8 in general already has that performance boost

**Comment by DPxKnightxZero** [comment]
Not really 1.8 I can actually see it for myself 1.8.9 has more than 1.8. You could always go see the change log for yourself : D

**Comment by luacs1998** [comment]
Well, most of the mods are skipping 1.8.0 and going up to 1.8.9 - I don't see why we should stay.

I'll do the port this weekend if time allows,

**Comment by DPxKnightxZero** [comment]
Thank you <3

**Comment by luacs1998** [comment]
@olee since I haven't be able to catch you on IRC:

Since the only broken parts of the 1.8.9 port are on the clientside mod, do you want to push server builds of FE 1.8.9 for people to use first? Then we can worry about fixing the client later.

**Comment by seanamor** [comment]
Absolutely dying to see 1.8.9 in action, after all its "essential" :P to minecraft! Cheer on all your hard work! my latest SpongeForge server will do well with such a masterpiece, not sure essentialcmds will suffice, lacking worldguard/perm support.

**Comment by luacs1998** [comment]
#2115
